ndcube/ndcube.py

```diff
@@ -405,7 +405,7 @@ def crop_by_coords(self, min_coord_values, interval_widths):
         # The docstring is defined in NDDataBase
 
         n_dim = len(self.dimensions)
-        if len(min_coord_values) != len(interval_widths) != n_dim:
+        if (len(min_coord_values) != len(interval_widths)) or len(min_coord_values) != n_dim:
```

Comment:
What's the difference in the logic of these two implementations?

Reply:
one works the other dosen't :laughing: 

I can't immediately work out why the first one dosen't work

Reply:
But ```NDCube.dimensions``` doesn't crash.  Do you mean that this if statement will never be entered or at least isn't entered when it should?

Reply:
```python
>>> d1 = 3
>>> d2 = 3
>>> nd = 4
>>> d1 != d2 != nd
False
>>> nd = 3
>>> d1 != d2 != nd
```

Reply:
Bizarre!